src/redux/businessPages/businessPages.sagas.js

```diff
@@ -0,0 +1,135 @@
+import { takeEvery, call, put } from 'redux-saga/effects';
+import { push } from 'connected-react-router';
+
+import {
+  setBusinessPages,
+  setLoading,
+  setBusinessPagesError,
+  setCurrentBusinessPage
+} from './businessPages.actions';
+import {
+  getAllBusinessPages,
+  getBusinessPageById,
+  createBusinessPage,
+  deleteBusinessPage,
+  updateBusinessPage
+} from './business-pages.operations';
+import {
+  ADD_BUSINESS_PAGE,
+  DELETE_BUSINESS_PAGE,
+  GET_ALL_BUSINESS_PAGES,
+  GET_BUSINESS_PAGE_BY_ID,
+  UPDATE_BUSINESS_PAGE
+} from './businessPages.types';
+
+import { config } from '../../configs';
+
+import {
+  setSnackBarSeverity,
+  setSnackBarStatus,
+  setSnackBarMessage
+} from '../snackbar/snackbar.actions';
+
+const {
+  SUCCESS_ADD_STATUS,
+  SUCCESS_DELETE_STATUS,
+  SUCCESS_UPDATE_STATUS
+} = config.statuses;
+
+export function* handleBusinessPagesLoad() {
+  try {
+    yield put(setLoading(true));
+    const businessPages = yield call(getAllBusinessPages);
+    yield put(setBusinessPages(businessPages));
+
+    yield put(setLoading(false));
+  } catch (error) {
+    yield call(handleBusinessPageError, error);
+  }
+}
+
+function* handleCurrentBusinessPageLoad({ payload }) {
+  try {
+    yield put(setLoading(true));
+
+    const businessPage = yield call(getBusinessPageById, payload);
+    yield put(setCurrentBusinessPage(businessPage));
+
+    yield put(setLoading(false));
+  } catch (error) {
+    yield call(handleBusinessPageError, error);
+  }
+}
+
+function* handleAddBusinessPage({ payload }) {
+  try {
+    yield put(setLoading(true));
+    yield call(createBusinessPage, payload);
+
+    const businessPages = yield call(getAllBusinessPages);
+    yield put(setBusinessPages(businessPages));
+
+    yield put(setSnackBarSeverity('success'));
+    yield put(setSnackBarMessage(SUCCESS_ADD_STATUS));
+    yield put(setSnackBarStatus(true));
+
+    yield put(setLoading(false));
+    yield put(push('/business-pages'));
+  } catch (error) {
+    yield call(handleBusinessPageError, error);
+  }
+}
+
+function* handleBusinessPageDelete({ payload }) {
+  try {
+    yield put(setLoading(true));
+    yield call(deleteBusinessPage, payload);
+
+    const businessPages = yield call(getAllBusinessPages);
+    yield put(setBusinessPages(businessPages));
+
+    yield put(setSnackBarSeverity('success'));
+    yield put(setSnackBarMessage(SUCCESS_DELETE_STATUS));
+    yield put(setSnackBarStatus(true));
+
+    yield put(setLoading(false));
+  } catch (error) {
+    yield call(handleBusinessPageError, error);
+  }
+}
+
+function* handleBusinessPageUpdate({ payload }) {
+  const { id, newPage } = payload;
+  try {
+    yield put(setLoading(true));
+    yield call(updateBusinessPage, id, newPage);
+
+    const businessPages = yield call(getAllBusinessPages);
```

Comment:
There is no need to fetch again business pages after creation, you can just update in the state with update data of the business page and avoid unnecessary HTTP call